pkg/security/utils/cgroup.go

```diff
@@ -289,7 +289,13 @@ func (cfs *CGroupFS) FindCGroupContext(tgid, pid uint32) (containerutils.Contain
 func checkPidExists(sysFScGroupPath string, expectedPid uint32) (bool, error) {
 	data, err := os.ReadFile(filepath.Join(sysFScGroupPath, "cgroup.procs"))
 	if err != nil {
-		return false, err
+		// the cgroup is in threaded mode, and in that case, reading cgroup.procs returns ENOTSUP.
+		// see https://github.com/opencontainers/runc/issues/3821
+		if errors.Is(err, unix.ENOTSUP) {
+			if data, err = os.ReadFile(filepath.Join(sysFScGroupPath, "cgroup.threads")); err != nil {
+				return false, err
+			}
+		}
```

Comment:
```suggestion
		} else {
		    return false, err
		}
```

Reply:
or use `if !errors.Is(err, unix.ENOTSUP)` instead to make the code more readable

Reply:
Good catch, thanks. Fixed